File: level2/ex54.js
function solution(land) {
    const n = land.length;
    const m = land[0].length;
    const arr = Array.from({ length: n}, () => Array(m).fill(0));
    
    for (let i = 0; i < n; i++) {
        for (let j = 0; j < m; j++) {
            if (i === 0) {
                arr[i][j] = land[i][j];
            } else {
                for (let k = 0; k < m; k++) {
                    if (k !== j) {
                        arr[i][j] = Math.max(arr[i][j], land[i][j] + arr[i-1][k]);
                    }
                }
            }
        }
    }
    
    return arr[n-1].reduce((a, b) => Math.max(a, b));
}

File: level2/ex54.py
def solution(land):
    n, m = len(land), len(land[0])
    arr = [[0] * m for i in range(n)]

    for i in range(n):
        for j in range(m):
            if i == 0:
                arr[i][j] = land[i][j]
            else:
                for k in range(m):
                    if k != j:
                        arr[i][j] = max(arr[i][j], land[i][j] + arr[i-1][k])
    
    return max(arr[-1])